vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -89,24 +95,28 @@ type Refreshable<'T> = 'T * (bool -> 'T)
 // The main entrypoints are TryGetOptionsForDocumentOrProject and TryGetOptionsForEditingDocumentOrProject.
 
 
-[<Export(typeof<FSharpProjectOptionsManager>); Composition.Shared>]
-type internal FSharpProjectOptionsManager 
+[<Export(typeof<ProjectInfoManager>); Composition.Shared>]
+type internal ProjectInfoManager 
     [<ImportingConstructor>]
     (
         checkerProvider: FSharpCheckerProvider,
         [<Import(typeof<SVsServiceProvider>)>] serviceProvider: System.IServiceProvider
     ) =
-    // A table of information about projects, excluding single-file projects.  
+    // A table of information about projects, excluding single-file projects.
     let projectTable = ConcurrentDictionary<ProjectId, Refreshable<ProjectId[] * FSharpProjectOptions>>()
 
     // A table of information about single-file projects.  Currently we only need the load time of each such file, plus
     // the original options for editing
     let singleFileProjectTable = ConcurrentDictionary<ProjectId, DateTime * FSharpProjectOptions>()
 
+    // Accumulate sorces and references for each project file
+    let projectInfo = new ConcurrentDictionary<string, string[]*string[]*string[]>()
+
     /// Clear a project from the project table
     member this.ClearInfoForProject(projectId: ProjectId) =
         projectTable.TryRemove(projectId) |> ignore
         this.RefreshInfoForProjectsThatReferenceThisProject(projectId)
+        //@@@@@@@ sourceFiles / referenceFiles / commandlineOptions
```

Comment:
?

Reply:
@@@@@@ is a mnemonic to me that I need to review this before I commit.  It is an incredibly old habit of mine. When I say old, I mean from the very early 80's.